surfactant/infoextractors/js_file.py

```diff
@@ -67,3 +73,85 @@ def match_by_attribute(attribute: str, content: str, database: Dict) -> List[Dic
                         # skip remaining patterns, move on to the next library
                         break
     return libs
+
+
+def download_database() -> dict:
+    url = "https://raw.githubusercontent.com/RetireJS/retire.js/master/repository/jsrepository-master.json"
+    response = requests.get(url)
+    if response.status_code == 200:
+        click.echo("Request successful!")
+        return json.loads(response.text)
+
+    if response.status_code == 404:
+        click.echo("Resource not found.")
+    else:
+        click.echo("An error occurred.")
+
+    return None
+
+
+def strip_irrelevant_data(retirejs_db: dict) -> dict:
+    clean_db = {}
+    reg_temp = "\u00a7\u00a7version\u00a7\u00a7"
+    version_regex = r"\d+(?:\.\d+)*"
+    for library, lib_entry in retirejs_db.items():
+        if "extractors" in lib_entry:
+            clean_db[library] = {}
+            patterns = lib_entry["extractors"]
+            possible_entries = [
+                "filename",
+                "filecontent",
+                "hashes",
+            ]
+            for entry in possible_entries:
+                if entry in patterns:
+                    entry_list = []
+                    for reg in patterns[entry]:
+                        entry_list.append(reg.replace(reg_temp, version_regex))
+                    clean_db[library][entry] = entry_list
+    return clean_db
+
+
+@surfactant.plugin.hookimpl
+def update_db():
+    """Retrieves the javascript library CVE database used by retire.js (https://github.com/RetireJS/retire.js/blob/master/repository/jsrepository-master.json) and only keeps the contents under each library's "extractors" section, which contains file hashes and regexes relevant for detecting a specific javascript library by its file name or contents.
+
+    The resulting smaller json is written to js_library_patterns.json in the same directory. This smaller file will be read from to make the checks later on."""
+    retirejs = download_database()
+    if retirejs is not None:
+        cleaned = strip_irrelevant_data(retirejs)
+        path = ConfigManager().get_data_dir_path() / "infoextractors"
+        path.mkdir(parents=True, exist_ok=True)
+        json_file_path = (
+            ConfigManager().get_data_dir_path() / "infoextractors" / "js_library_patterns.json"
+        )
+        with open(json_file_path, "w") as f:
+            json.dump(cleaned, f, indent=4)
+        return "Update complete."
+    return "No update occurred."
+
+
+@surfactant.plugin.hookimpl
+def short_name():
+    return "js_file"
+
+
+def load_db():
+    js_lib_file = (
+        ConfigManager().get_data_dir_path() / "infoextractors" / "js_library_patterns.json"
+    )
+
+    try:
+        path = ConfigManager().get_data_dir_path() / "infoextractors"
+        path.mkdir(parents=True, exist_ok=True)
```

Comment:
This can be removed, since the directory will exist after the user runs:
`surfactant plugin update-db <js_file or --all>`